syntax = "proto3";
package ledger.core.message;

enum CoreRequestType {
    GET_VERSION = 0;
    SET_CONFIGURATION = 1;
    BITCOIN_REQUEST = 2;
    ETHEREUM_REQUEST = 3;
    XRP_REQUEST = 4;
}

message CoreRequest {
    CoreRequestType request_type = 1;
    bytes request_body = 2;
}

message CoreResponse {
    bytes response_body = 1;
    string error = 2;
}

message GetVersionResponse {
    int32 major = 1;
    int32 minor = 2;
    int32 patch = 3;
}
